primitives/src/communities.rs

```diff
@@ -50,6 +55,48 @@ pub fn validate_nominal_income(nominal_income: &NominalIncome) -> Result<(), ()>
     Ok(())
 }
 
+fn fmt(cid: &CommunityIdentifier, f: &mut Formatter<'_>) -> fmt::Result {
+    write!(f, "{}", bs58::encode(cid.as_array()).into_string())
+}
+
+impl fmt::Display for CommunityIdentifier {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        fmt(self, f)
+    }
+}
+
+impl fmt::Debug for CommunityIdentifier {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        fmt(self, f)
+    }
+}
+
+impl CommunityIdentifier {
+    pub fn new<AccountId: Encode>(location: Location, bootstrappers: Vec<AccountId>) -> CommunityIdentifier {
+        let mut geohash_cropped = [0u8; 5];
+        geohash_cropped.clone_from_slice(&GeoHash::try_from_params(location.lat, location.lon, 5).unwrap()[0..5]);
```

Comment:
Can we be certain that this does not fail? Runtime code must not panic under any circumstances.

Reply:
When we have resolved potential invalid lon/lat, I would like the unwrap to be an `.expect("Location struct can't contain invalid lon/lat; qeg");`

Reply:
we need to remove the unwrap for sure. But I don't get why we need to validate the location before generating the geohash. geohash creation should fail anyway if locations are invalid. I think we do double work here

Reply:
> we need to remove the unwrap for sure. But I don't get why we need to validate the location before generating the geohash. geohash creation should fail anyway if locations are invalid. I think we do double work here

now you got me confused, as this is - as I understood - what I asked in my last post, whether you are sure you want me to implement this..

Reply:
ah, wou would want to map the result of try_from _params into the communityIdentifierError?